processing/src/main/java/org/apache/druid/query/groupby/GroupByQuery.java

```diff
@@ -577,6 +577,7 @@ private Ordering<ResultRow> getRowOrderingForPushDown(
     final List<Boolean> needsReverseList = new ArrayList<>();
     final List<ColumnType> dimensionTypes = new ArrayList<>();
     final List<StringComparator> comparators = new ArrayList<>();
+    final List<DimensionSpec> dimensionsInOrder = new ArrayList<>();
```

Comment:
this double-book-keeping is a pretty wierd way to supply this detail - but it works...

I think it would be more beneficial
* identify the differences between `getRowOrderingForPushDown` and `getRowOrdering` 
* merge the differing behaviour into the `getRowOrderingForPushDown` method
   * isn't the main difference is that `getRowOrdering` ignores the `limitSpec` regardless its set?
* possibly also eat-up the `RowBasedGrouperHelper` which is another copy of the process of creating the comparators - I think correctness of the execution depends on knowing on which columns we ordered and in what order....having just one source of truth could reduce the amount of issues we face
 
all of the above are refactors - which might delay the fix of this bug....